libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Memory/Expressions/ValueExpression.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.Bot.Builder.Dialogs.Adaptive.Converters;
+using Microsoft.Bot.Builder.LanguageGeneration;
+using Microsoft.Bot.Expressions;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Bot.Builder.Dialogs.Adaptive
+{
+    /// <summary>
+    /// ValueExpression - represents a property which is an object of any kind or a string expression.
+    /// </summary>
+    /// <remarks>
+    /// If the value is 
+    /// * a string with '=' prefix then the string is treated as an expression to resolve to a string. 
+    /// * a string without '=' then value is treated as string with string interpolation.
+    /// * any other type, then it is of that type (int, bool, object, etc.)
+    /// You can escape the '=' prefix by putting a backslash.  
+    /// Examples: 
+    ///     prop = true ==> true
+    ///     prop = "Hello @{user.name}" => "Hello Joe"
+    ///     prop = "=length(user.name)" => 3
+    ///     prop = "=user.age" => 45.
+    ///     prop = "\=user.age" => "=user.age".
+    /// </remarks>
+    [JsonConverter(typeof(ValueExpressionConverter))]
+    public class ValueExpression : ExpressionProperty<object>
+    {
+        private LGFile lg = new LGFile();
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ValueExpression"/> class.
+        /// </summary>
+        public ValueExpression()
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ValueExpression"/> class.
+        /// </summary>
+        /// <param name="value">value to interpret as object or string expression.</param>
+        public ValueExpression(object value)
+            : base(value)
+        {
+        }
+
+        public static implicit operator ValueExpression(string valueOrExpression) => new ValueExpression(valueOrExpression);
```

Comment:
>public static [](start = 8, length = 13)

Do we not have to document public statics?  Missing here and a couple of other places.